python/src/trezorlib/device.py

```diff
@@ -79,23 +79,34 @@ def apply_settings(
         haptic_feedback=haptic_feedback,
     )
 
-    out = client.call(settings, expect=messages.Success)
+    if homescreen is not None:
+        if client.version < HOMESCREEN_STREAMING_MIN_VERSION:
+            settings.homescreen = homescreen
+        else:
+            settings.homescreen_length = len(homescreen)
+
+    response = client.call(settings)
+    if settings.homescreen_length is not None:
+        response = _send_chunked_data(client, response, homescreen or b"")
+    else:
+        messages.Success.ensure_isinstance(response)
     client.refresh_features()
-    return _return_success(out)
+    return _return_success(response)
```

Comment:
instead something vaguely like this:
```
if homescreen is not None and client.version >= HOMESCREEN:
    settings.homescreen_length = len(homescreen)
    client.call(settings, expect=DataChunkRequest)
    _send_chunked_data(client, response, homescreen)
    return _return_success(messages.Success(""))
else:
   out = client.call(settings, expect=Success)
   return _return_success(out)
```
(IIUC the purpose to returning Success from `_send_chunked_data` was to propagate it here, correct? let's not do that)

Reply:
fd1419a708062560e063c3d26ac3e20da93520ac